tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -654,9 +762,27 @@ void TangoRosNode::PublishColorImage() {
       std::unique_lock<std::mutex> lock(color_image_available_mutex_);
       color_image_available_.wait(lock);
       if ((publisher_config_.publish_camera & CAMERA_COLOR)) {
-        compressImage(color_image_, CV_IMAGE_COMPRESSING_FORMAT,
-                      IMAGE_COMPRESSING_QUALITY, &color_compressed_image_);
-        color_image_publisher_.publish(color_compressed_image_);
+        // The Tango image encoding is not supported by ROS.
+        // We need to convert it to RGB.
+        cv::Mat color_image_rgb;
+        cv::cvtColor(color_image_, color_image_rgb, cv::COLOR_YUV420sp2BGRA);
+        cv_bridge::CvImage cv_bridge_image;
+        cv_bridge_image.header = color_image_header_;
+        cv_bridge_image.encoding = sensor_msgs::image_encodings::BGRA8;
+        cv_bridge_image.image = color_image_rgb;
+        color_camera_info_.header = color_image_header_;
+        color_camera_info_manager_->setCameraInfo(color_camera_info_);
+        color_camera_publisher_.publish(*(cv_bridge_image.toImageMsg()), color_camera_info_);
+
+        if (color_rectified_image_publisher_.getNumSubscribers() > 0) {
```

Comment:
This can be done in another PR. It is not that straightforward because it means removing the publish_* parameters (from dynamic reconfigure and from the app). Issue #60 is already mentioning this.